doc/user/content/cli/_index.md

```diff
@@ -200,3 +202,58 @@ etc.), and then create a new node with those items.
 [gh-feature]: https://github.com/MaterializeInc/materialize/issues/new?labels=C-feature&template=feature.md
 [scv]: /sql/show-create-view
 [scs]: /sql/show-create-source
+
+### Persistence
+
+{{< experimental v0.4.2 >}}
+
+To avoid re-reading data from Kafka on restart, Materialize lets you create
+persistent sources, which persist input messages from Kafka topics to files
+on the Materialize instance's local hard drive. Persistence will not solve all
+restart time related problems, as there are other factors beyond Kafka broker
+read performance that contribute to high restart times.
+
+We recommend enabling persistence if you are using Kafka sources, need to relieve
+load on upstream Kafka brokers, and are comfortable using
+[experimental](#experimental-mode) features.
+
+{{< warning >}}
+
+Materialize currently does not delete persisted records when the source is dropped.
+Additionally, Materialize does not currently compact persisted data. If you enable
+persistence on sources from compacted Kafka topics, Materialize will store and re-read
+all records that have been persisted, even if some of them were compacted by the upstream
+source.
+
+{{< /warning >}}
+
+#### Details
+
+You can define a persistent source with the [`CREATE SOURCE`][cs] command. Materialize
+stores one copy of all input data for each persistent Kafka source. Materialize
+stores these files in:
+
+```
+{data-directory}/persistence/{source-id}
+```
+
+Within this directory, Materialize writes to files named
+
+```
+materialize-{source-id}-{partition-id}-{start-offset}-{end-offset}
+```
+
+Here, each file stores data for ranges of offsets per `partition-id`. Each file
+stores all the data from `start-offset` (inclusive) to `end-offset` (exclusive).
+Materialize buffers up to `--persistence-max-pending-records` records in memory
+per source, before flushing them all to disk immediately. Setting this flag to a
+higher value helps Materialize achieve higher ingest and disk write throughput,
+however this also increases the average latency before records are persisted.
+Additionally, Materialize flushes input records to disk every 10 minutes.
+
+
+On restart, Materialize reads back all of the records that had been previously
+persisted in offset order, and then continues reading from the upstream source
+for data after the last persisted record in each partition.
+
+[cs]: /sql/create-source
```

Comment:
you can link this to the anchor tag for the example you addd

Reply:
if i add an example it has to be one specific source type right (like kafka + avro)

so should i do that and then just add a link here (even though it doesnt cover all source types) or should i add an example for each source type?

Reply:
We should add an example for each source type that supports persistence; if there's more than one then maybe consider adding a parenthetical with the links or something